Feature/doc updates chef solo

dereklwood opened the pull request:
chef-solo-automator doc updates

Commit:
coopr_service_runner renamed to coopr_service_manager

Commit:
documenting helper cookbooks and other chef-solo chgs for 0.9.9

awholegunch commented on coopr-docs/docs/source/guide/superadmin/chef-solo-automator-plugin.rst:
```diff
@@ -141,25 +139,34 @@ service-level action data are merged together, and preserved at the top-level. C
+		c. via the Opscode Omnibus installer (``curl -L https://www.opscode.com/chef/install.sh | bash``). This requires internet access on the remote host
+	3. Creates the remote coopr cache directory ``/var/cache/coopr``
+	4. SCP the local tarballs to the remote Coopr cache directory
+        5. Extracts the tarballs on the remote host to the default chef directory ``/var/chef``
```
remove a space before the "5" so that it lines up, otherwise it won't format correctly. ProTip: you can do the first item as  "1." and then the rest as "#.", and Sphinx will number them for you accordingly. As long as they line up. Handy if you add/subtract/move things...

awholegunch commented on coopr-docs/docs/source/guide/superadmin/chef-solo-automator-plugin.rst:
```diff
@@ -141,25 +139,34 @@ service-level action data are merged together, and preserved at the top-level. C
+	1. Chef must be pre-installed on the target hosts' images, or installable via a pre-configured Yum or Apt repository
+	2. The bundled community cookbooks cannot be used.  Note the provided `Helper Cookbooks`_ do not require internet access
 
 Adding your own Cookbooks
```
Adding Your Own Cookbooks (keeping capitalization consistent)

awholegunch commented on coopr-docs/docs/source/guide/superadmin/chef-solo-automator-plugin.rst:
```diff
@@ -198,22 +211,77 @@ definition with the following parameters:
 	* run ``apt-get update`` (on Ubuntu hosts)
-	* include ``coopr_hosts::default`` (discussed below)
-	* include ``coopr_firewall::default`` (discussed below)
+        * optionally configure the Yum ``epel`` repository (on Rhel hosts)
```
remove leading space

awholegunch commented on coopr-docs/docs/source/guide/superadmin/chef-solo-automator-plugin.rst:
```diff
@@ -198,22 +211,77 @@ definition with the following parameters:
+	* optionally include ``coopr_hosts::default`` (discussed below)
+	* optionally include ``coopr_packages::default`` (discussed below)
 	* include ``ulimit::default`` to enable user-defined ulimits
+        * optionally runs the ``users`` and ``sudo`` cookbooks to add any users if a ``users`` data_bag resource is present
```
remove leading space

awholegunch commented on coopr-docs/docs/source/guide/superadmin/chef-solo-automator-plugin.rst:
```diff
@@ -198,22 +211,77 @@ definition with the following parameters:
 	* include ``ulimit::default`` to enable user-defined ulimits
+        * optionally runs the ``users`` and ``sudo`` cookbooks to add any users if a ``users`` data_bag resource is present
+
+To disable the configuring of the Yum ``epel`` repository on Rhel hosts, set the ``node['base']['use_epel']`` attribute to ``false``
```
add trailing period.

awholegunch commented on coopr-docs/docs/source/guide/superadmin/chef-solo-automator-plugin.rst:
```diff
@@ -198,22 +211,77 @@ definition with the following parameters:
+
+To disable the configuring of the Yum ``epel`` repository on Rhel hosts, set the ``node['base']['use_epel']`` attribute to ``false``
+
+To disable the inclusion of the remaining helper cookbooks, set the following attributes. Though note that these cookbooks are designed to not take action unless
```
Note, though, that these cookbooks...

awholegunch commented on coopr-docs/docs/source/guide/superadmin/chef-solo-automator-plugin.rst:
```diff
@@ -198,22 +211,77 @@ definition with the following parameters:
+    node['base']['no_hosts'] = 'true'
+    node['base']['no_packages'] = 'true'
+
+**User Management**
```
Did you want underline under this to make a section?

dereklwood replied:
it's part of the coopr_base cookbook, so i think it should be a sub-section.  If there is a better format to use, let me know

awholegunch replied:
Use a rubric then, as that creates a divider without creating a new section:

```
.. rubric:: User Management
```

awholegunch commented on coopr-docs/docs/source/guide/superadmin/chef-solo-automator-plugin.rst:
```diff
@@ -198,22 +211,77 @@ definition with the following parameters:
+
+ /opt/coopr/provisioner/embedded/bin/ruby /opt/coopr/provisioner/bin/data-uploader.rb -u |http:|//localhost:55054 -t superadmin -U admin sync ./my/local/data_bags/users automatortypes/chef-solo/data_bags/users
+
+where ``./my/local/data_bags/users`` is the data_bag to be used by the ``users`` cookbook. It should be a directory containing data_bag_items (json files) of the form::
```
JSON

awholegunch commented on coopr-docs/docs/source/guide/superadmin/chef-solo-automator-plugin.rst:
```diff
@@ -198,22 +211,77 @@ definition with the following parameters:
+
+    {
+        "id": "bob",
+        "Full Name": "Bob User",
```
is "Full Name" correct as a key? With a space in it?

dereklwood replied:
doublechecked... yes

awholegunch commented on coopr-docs/docs/source/guide/superadmin/chef-solo-automator-plugin.rst:
```diff
@@ -198,22 +211,77 @@ definition with the following parameters:
+
+The ``provider`` attribute is simply the name of a recipe that must be implemented.  The ``subdomain_whitelist`` is an optional safety mechanism to only allow
+DNS records to be created for certain domains.  In this example, as part of the ``base`` service the ``coopr_dns::dnsimple`` recipe will be run, which includes
+DNSimple's cookbook and its LWRPs to create A records for each cluster host.
```
```
DNSimple's cookbook and its LWRPs to create "A" records for each cluster host.
```

duddun commented on coopr-docs/docs/source/guide/superadmin/chef-solo-automator-plugin.rst:
```diff
@@ -227,16 +295,18 @@ The example coopr service definition invoking this cookbook is called "base". It
 action of type "chef-solo" and run_list ``recipe[coopr_base::default]`` (which includes ``recipe[coopr_hosts::default]``).
 Note that the community "hostsfile" cookbook is not needed in the run-list since it is declared in coopr_hosts's metadata.
```
This should read 'run list' (I realize this is not part of the current changes)

dereklwood replied:
Chef docs use 'run-list' when referring to it in the general sense, which i tried to be consistent with.  But our json field is "run_list".  So something may still be incorrect but i'm not sure

awholegunch commented on coopr-docs/docs/source/guide/superadmin/chef-solo-automator-plugin.rst:
```diff
@@ -257,6 +326,18 @@ nodes of a cluster. To use, simply set any of the following attributes:
+**coopr_packages**
+------------------
+
+This cookbook provides a convient way to install, upgrade, or remove any number of Yum or Apt packages as part of your ``base`` service.  Simply populate
```
provides a convenient way

duddun replied:
change 'convient' to 'convenient'

awholegunch commented:
Bunch o' comments...

duddun commented on coopr-docs/docs/source/guide/superadmin/chef-solo-automator-plugin.rst:
```diff
@@ -227,16 +295,18 @@ The example coopr service definition invoking this cookbook is called "base". It
 cluster. It allows you to simply specify the name of a Chef service resource and an action within a Coopr service
 definition. When run, it will simply lookup the Chef service resource of the given name, regardless of which cookbook
 it is defined in, and run the given action. In the example apache-httpd service definition above, it is simply included
 in the run-list to start or stop the apache2 service defined in the apache2 community cookbook. All that is needed is
```
change 'run-list' to 'run list'

duddun replied:
Change 'All that is needed is to set the following' to 'Just set the following'

dereklwood replied:
i disagree with the run-list.  Chef docs use "run-list", ie http://docs.chef.io/nodes.html#about-run-lists

duddun commented on coopr-docs/docs/source/guide/superadmin/chef-solo-automator-plugin.rst:
```diff
@@ -227,16 +295,18 @@ The example coopr service definition invoking this cookbook is called "base". It
 This cookbook comes in handy as a simple way to isolate the starting and stopping of various services within your
 cluster. It allows you to simply specify the name of a Chef service resource and an action within a Coopr service
 definition. When run, it will simply lookup the Chef service resource of the given name, regardless of which cookbook
 it is defined in, and run the given action. In the example apache-httpd service definition above, it is simply included
```
change this to read '. In the apache-httpd service definition example above,'

duddun commented on coopr-docs/docs/source/guide/superadmin/chef-solo-automator-plugin.rst:
```diff
@@ -245,8 +315,7 @@ to set the following attribute to "start" or "stop":
 This cookbook is a simple iptables firewall manager, with the added functionality of automatically whitelisting all
-nodes of a cluster. To use, simply set any of the following attributes:
-::
+nodes of a cluster. To use, simply set any of the following attributes::
```
change to: 'all nodes in a cluster.'

duddun commented on coopr-docs/docs/source/guide/superadmin/chef-solo-automator-plugin.rst:
```diff
@@ -257,6 +326,18 @@ nodes of a cluster. To use, simply set any of the following attributes:
 
 If this recipe is included in the run-list and no attributes specified, the default behavior is to disable the firewall.
```
change 'run-list' to 'run list'

duddun commented on coopr-docs/docs/source/guide/superadmin/chef-solo-automator-plugin.rst:
```diff
@@ -40,8 +40,7 @@ chef-solo run is self-contained and intended to perform a specific task such as
 the typical usage of Chef where one builds up a large run-list managing all resources on a box in one run.
```
change 'run-list' to 'run list'

duddun commented on coopr-docs/docs/source/guide/superadmin/chef-solo-automator-plugin.rst:
```diff
@@ -119,13 +118,12 @@ Coopr maintains significant JSON data for a cluster, and makes it available for
-``coopr/*``. For example:
-::
+service-level action data are deep-merged together, with service-level action data taking precedence. This data is
+preserved at the top-level, and also merged in under ``coopr/*``. For example::
```
change 'top-level' to 'top level'

duddun commented on coopr-docs/docs/source/guide/superadmin/chef-solo-automator-plugin.rst:
```diff
@@ -141,25 +139,34 @@ service-level action data are merged together, and preserved at the top-level. C
-Consider the following two rules of thumb:
-	* When using community cookbooks, attributes can be specified in Coopr templates exactly as the cookbook expects (at the top-level).
+Consider the following rules of thumb:
+	* When using community cookbooks, attributes can be specified in Coopr templates exactly as the cookbook expects (at the top-level). If separate services require the same recipe with different attributes, then these attributes can be specified in the service ``json_attributes`` field
```
change 'top-level' to 'top level'

duddun replied:
'then' is unnecessary here

duddun commented on coopr-docs/docs/source/guide/superadmin/chef-solo-automator-plugin.rst:
```diff
@@ -141,25 +139,34 @@ service-level action data are merged together, and preserved at the top-level. C
 Bootstrap
 =========
 
 Each Coopr Automator plugin is responsible for implementing a bootstrap method in which it performs any actions it needs to be able to carry out further tasks. The Chef Solo Automator plugin performs the following actions for a bootstrap task:
```
use 'any action' or 'all actions'

duddun commented on coopr-docs/docs/source/guide/superadmin/chef-solo-automator-plugin.rst:
```diff
@@ -141,25 +139,34 @@ service-level action data are merged together, and preserved at the top-level. C
-	3. Creates the remote coopr cache directory ``/var/cache/coopr``.
-	4. SCP the local tarballs to the remote Coopr cache directory.
-        5. Extracts the tarballs on the remote host to the default chef directory ``/var/chef``.
+	1. Bundle its local copy of the cookbooks/roles/data_bags directories into tarballs, ``cookbooks.tar.gz``, ``roles.tar.gz``, ``data_bags.tar.gz``
```
change this to: 
[...]directories into three tarballs: `cookbooks.tar.gz`, `roles.tar.gz`, and `data_bags.tar.gz`.

duddun commented on coopr-docs/docs/source/guide/superadmin/chef-solo-automator-plugin.rst:
```diff
@@ -141,25 +139,34 @@ service-level action data are merged together, and preserved at the top-level. C
-	4. SCP the local tarballs to the remote Coopr cache directory.
-        5. Extracts the tarballs on the remote host to the default chef directory ``/var/chef``.
+	1. Bundle its local copy of the cookbooks/roles/data_bags directories into tarballs, ``cookbooks.tar.gz``, ``roles.tar.gz``, ``data_bags.tar.gz``
+		* Unless the tarballs exist already and were created in the last 10 minutes
```
Change to: \* This only happens if prior versions of the tarballs were not created in the past 10 minutes.

duddun commented on coopr-docs/docs/source/guide/superadmin/chef-solo-automator-plugin.rst:
```diff
@@ -141,25 +139,34 @@ service-level action data are merged together, and preserved at the top-level. C
+		b. via ``apt-get install`` to leverage any internal apt repositories preconfigured on the remote host
+		c. via the Opscode Omnibus installer (``curl -L https://www.opscode.com/chef/install.sh | bash``). This requires internet access on the remote host
+	3. Creates the remote coopr cache directory ``/var/cache/coopr``
+	4. SCP the local tarballs to the remote Coopr cache directory
```
Change 'SCP' to 'SCPs' for consistency with the other four numbered actions.

duddun commented on coopr-docs/docs/source/guide/superadmin/chef-solo-automator-plugin.rst:
```diff
@@ -141,25 +139,34 @@ service-level action data are merged together, and preserved at the top-level. C
+	4. SCP the local tarballs to the remote Coopr cache directory
+        5. Extracts the tarballs on the remote host to the default chef directory ``/var/chef``
 
 The most important things to note are that:
```
remove 'are that'.

Commit:
fix alignment and use auto-numbering

Commit:
corrections per PR feedback

Commit:
convert all tabs to spaces

awholegunch commented:
One comment. otherwise LGTM...

Commit:
use rubric to divide User Management without creating a separate section

dereklwood commented:
I've incorporated all feedback, except I'm keeping it "run-list" since that what Chef uses.  And I converted all tabs to spaces per @awholegunch , which might've thrown off github's diff/comment association.
